custom_components/dualmode_generic/climate.py

```diff
@@ -330,17 +358,29 @@ async def async_set_hvac_mode(self, hvac_mode):
             self._hvac_mode = HVAC_MODE_HEAT
             if self._is_device_active and not self.reverse_cycle:
                 await self._async_cooler_turn_off()
+                if self.fan_entity_id is not None:
```

Comment:
You could put this `if` test in the `_async_fan_turn_off` function to avoid having to repeat it here and below